modules/basicfuncs/fname-funcs.c

```diff
@@ -30,7 +30,7 @@ tf_basename(LogMessage *msg, gint argc, GString *argv[], GString *result)
   gchar *base;
 
   base = g_path_get_basename(argv[0]->str);
-  g_string_assign(result, base);
+  g_string_append(result, base);
```

Comment:
Just for the reference:
https://github.com/balabit/syslog-ng/blob/da326364d52a21ebaf2d8fd17f4077fc5cc954e6/lib/template/templates.c#L170

For me this was not trivial why we can eliminate the initialization of the result variable. Seems like it was unnecessary in the first place.